template/src/steps/fetchAccounts.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  createIntegrationEntity,
+} from '@jupiterone/integration-sdk';
+
+const step: IntegrationStep = {
+  id: 'fetch-accounts',
+  name: 'Fetch accounts',
+  types: ['my_integration_account'],
+  async executionHandler({
+    logger,
+    jobState,
+  }: IntegrationStepExecutionContext) {
+    try {
+      await jobState.addEntities([
+        createIntegrationEntity({
+          entityData: {
+            source: {
+              id: 'integration-account-a',
+              name: 'My Account',
+            },
+            assign: {
+              _key: 'account:integration-account-a',
+              _type: 'my-integration-account',
+              _class: 'Account',
+            },
+          },
+        }),
+      ]);
+    } catch (err) {
+      logger.error(err, 'Unexpected error occurred');
+      throw err;
+    }
```

Comment:
I think we should encourage just allowing the error to be thrown (no `try/catch` needed), since the runtime environment will log the error. This can simplify steps a good deal as folks don't have to think about it unless they want to do something more than logging. This example suggests they must log errors themselves.